src/log.c

```diff
@@ -2,8 +2,91 @@
 
 #include <stdbool.h>
 #include <stdio.h>
+#include <time.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <string.h>
+
+#include "util.h"
 
 bool verbose;
+FILE *log_file;
+
+void construct_log_filename(char **log_filename, const char *log_filepath)
+{
+	time_t t = time(NULL);
+	struct tm tm = *localtime(&t);
+
+	sprintf(*log_filename, "%scnc%02d%02d%02d_%02d%02d%02d.log",
+		log_filepath, tm.tm_mday, tm.tm_mon + 1, tm.tm_year + 1900,
+		tm.tm_hour, tm.tm_min, tm.tm_sec);
+}
+
+/*
+ *  construct_log_filepath()
+ *
+ *  Returns:
+ *  0 Success
+ * -3 Failure
+ */
```

Comment:
Move the doc-comment to log.h. I think we have it like that for other functions too.